source/extensions/filters/http/basic_auth/config.cc

```diff
@@ -0,0 +1,69 @@
+#include "source/extensions/filters/http/basic_auth/config.h"
+
+#include "source/common/config/datasource.h"
+#include "source/extensions/filters/http/basic_auth/basic_auth_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace BasicAuth {
+
+using envoy::extensions::filters::http::basic_auth::v3::BasicAuth;
+
+namespace {
+
+UserMap readHtpasswd(std::string htpasswd) {
+  UserMap users;
+  std::istringstream htpsswd_ss(htpasswd);
+  std::string line;
+
+  while (std::getline(htpsswd_ss, line)) {
+    size_t colonPos = line.find(':');
+
+    if (colonPos != std::string::npos) {
+      std::string name, hash;
+
+      name = line.substr(0, colonPos);
+      hash = line.substr(colonPos + 1);
+
+      if (name.length() == 0) {
+        throw EnvoyException("invalid user name");
+      }
```

Comment:
`name.empty()` (clang tidy will complain this. Although we haven't fixed our clang tidy ci.)


`throw EnvoyException("basic auth: invalid user name");`